api/command/mixins/team.py

```diff
@@ -0,0 +1,701 @@
+                 or specified team cannot be found in the database
+        :raises: RuntimeError if more than one team has the specified
+                 team name
+        :raises: PermissionError if the calling user has insufficient
+                 permission to add members to the specified team
+        :raises: GithubAPIException if an error occurs when adding the user to
+                 the Github team
+        :return: True if adding the member to the team was successful,
+                 False otherwise
+        """
+        logging.info("Team add command API called")
+        try:
+            command_user = self._db_facade.retrieve(User, user_id)
+            logging.debug(f"Calling user: {command_user.__str__()}")
+        except LookupError:
+            msg = f"Calling user with Slack ID {user_id} not found!"
+            logging.error(msg)
+            raise LookupError(msg)
+
+        teams = self._db_facade.query(Team,
+                                      [('github_team_name', gh_team_name)])
+        num_teams = len(teams)
+
+        if num_teams < 1:
+            msg = f"No teams found with team name {gh_team_name}"
+            logging.error(msg)
+            raise LookupError(msg)
+        elif num_teams > 1:
+            msg = f"{num_teams} found with team name {gh_team_name}"
+            logging.error(msg)
+            raise RuntimeError(msg)
+        else:
+            team = teams[0]
+            logging.info(f"Team queried with team name {gh_team_name}:"
+                         f" {team.__str__()}")
+
+        if not check_permissions(command_user, team):
+            msg = f"User with ID {user_id} has insufficient permissions to " \
+                f"add members to team {gh_team_name}"
+            logging.error(msg)
+            raise PermissionError(msg)
+
+        try:
+            add_user = self._db_facade.retrieve(User, slack_id)
+            logging.debug(f"User to add: {add_user.__str__()}")
+        except LookupError:
+            msg = f"User to add with Slack ID {slack_id} not found!"
+            logging.error(msg)
+            raise LookupError(msg)
+
+        try:
+            self._gh_interface.add_team_member(add_user.github_username,
+                                               team.github_team_id)
+        except GithubAPIException as e:
+            msg = f"Error adding {add_user.github_username} to " \
+                f"Github team: {e.data}"
+            logging.error(msg)
+            raise GithubAPIException(msg)
+        team.add_member(add_user.github_id)
+
+        added = cast(bool, self._db_facade.store(team))
+        if added:
+            logging.info(f"Member successfully added: {team.__str__()}")
+        else:
+            logging.error("Member add unsuccessful")
+        return added
+
+    def team_remove(self,
+                    user_id: str,
+                    gh_team_name: str,
+                    slack_id: str) -> bool:

```

Comment:
Same here with unclear `user_id` and `slack_id`

Maybe have it called caller_id (caller is implicitly a user) or something.